pkg/coord/local/clocal.go

```diff
@@ -666,8 +704,13 @@ func (qr *LocalCoordinator) ListKeyRanges(ctx context.Context, distribution stri
 		return nil, err
 	} else {
 		for _, keyRange := range krs {
-			ret = append(ret, kr.KeyRangeFromDB(keyRange))
+			ds, err := qr.qdb.GetDistribution(ctx, keyRange.DistributionId)
```

Comment:
You can get distribution only once per function call

Reply:
si